arch/x86/x86_features.c

```diff
@@ -15,6 +15,13 @@
 #else
 // Newer versions of GCC and clang come with cpuid.h
 #  include <cpuid.h>
+#  ifdef X86_HAVE_XSAVE_INTRIN
+#    if __GNUC__ == 8
```

Comment:
Suggest:
```c
#    if defined(__GNUC__) && __GNUC__ == 8
```

And in other places.

Reply:
> Suggest:
> ```c
> #    if defined(__GNUC__) && __GNUC__ == 8
> ```
> 
> And in other places.

It's not necessary as we already rely on anything undefined to be interpreted as `0`.